crates/tdb-server-core/src/plugins/gstorage/upload.rs

```diff
@@ -69,27 +63,28 @@ impl GStorageFile {
             self.remote_name
         );
 
-        let body = self.file_content();
+        let file = File::open(&self.fname)?;
+
+        let client = reqwest::blocking::Client::new();
 
-        let client = reqwest::Client::new();
         let mut res = client
             .post(&uri)
-            .body(body?)
+            .body(file)
             .send()?;
 
         if res.status().is_success() {
             let mut content = String::new();
             let _ = res.read_to_string(&mut content);
 
             // get end time
-            let finish_ts = time::now();
+            let finish_ts = time::Instant::now().elapsed().whole_milliseconds();
 
             self.uploaded = true;
 
             Ok(GStorageOpMetadata::new(
                 content,
-                start_ts.to_timespec().sec as u32,
-                finish_ts.to_timespec().sec as u32,
+                start_ts as u32,
+                finish_ts as u32,
```

Comment:
I've changed seconds to milliseconds, as said in README in this folder

Reply:
Thanks for updating. This looks good.